### ISSUE TITLE
On Machine always runs commands as root

### ISSUE THREAD

**Issue by pizzaops** [oscar-stack/vagrant-pe_build#126]
See: https://github.com/oscar-stack/vagrant-pe_build/blob/48acc7abb3a3b9c38d4b14fc23df674a69f5eb8e/lib/pe_build/on_machine.rb

This is problematic because if pe_build tries to upload any files to directories created with on_machine, it fails, since vagrant scp runs as the non-sudo user.

In this case, it seems to be causing a failure when vagrant tries to "fix" the out of date GPG key shipped with some PE releases.

```
Source: /Users/zee/.vagrant.d/gems/gems/vagrant-pe_build-0.15.5/data/vagrant-pe_build/files/GPG-KEY-puppetlabs
Dest: puppet-enterprise-2015.3.2-el-6-x86_64/gpg/GPG-KEY-puppetlabs
ERROR vagrant: Catched Error: Failed to upload a file to the guest VM via SCP due to a permissions
error. This is normally because the SSH user doesn't have permission
to write to the destination location. Alternately, the user running
Vagrant on the host machine may not have permission to read the file
```

So in this case what's happening is:

1. The PE installer is staged on the machine, as root, via running commands with `on_machine`.
1. Vagrant/PE Build detects the release is 2015.3.2, which requires a GPG key update since the key originally shipped with that release has expired.
1. Vagrant/PE Build tries to SCP up the updated key, which fails, because it runs the SCP not-as-root, and it can't write to the directory containing the installer.

Update - worked around by adding sudo to these commands:

https://github.com/oscar-stack/vagrant-pe_build/blob/master/lib/pe_build/cap/run_install/posix.rb#L34-L40

**Comment by pizzaops** [comment]
Adding sudo did *not* fix it. I made a mistake. Not sure what you'll have to do, since I've got to fall back to virtualbox and get some stuff done :(

**Comment by Sharpie** [comment]
Just shipped 0.15.6 which contains a fix for this. Thanks for reporting it!
